src/foam/dao/MDAO.java

```diff
@@ -107,11 +107,37 @@ public MDAO(ClassInfo of) {
     index_ = new AltIndex(new TreeIndex((PropertyInfo) this.of_.getAxiomByName("id")));
   }
 
+
+
+  protected ArrayList<PropertyInfo> propertyInfoList_(Index index) {
+    ArrayList<PropertyInfo> infoList = new ArrayList<PropertyInfo>();
+    var current = index;
+    while ( current != null && ( current instanceof TreeIndex ) ) {
+      infoList.add(((TreeIndex)current).getProp());
+      current = ((TreeIndex)current).getTail();
+    }
+    return infoList;
+  }
+
   public void addIndex(Index index) {
     synchronized ( writeLock_ ) {
+      if ( index instanceof TreeIndex ) {
+        var properties = propertyInfoList_(index);
+        var delegates = index_.getDelegates();
+        for ( Index i : delegates ) {
+          var properties2 = propertyInfoList_(i);
+          if ( properties.containsAll(properties2) || properties2.containsAll(properties) ) {
```

Comment:
Better to check if one is a prefix of the other, since something like:
A, B, C
and A, C
might both be required.